src/WorkspaceRegistration/WorkspaceRegistration.tsx

```diff
@@ -0,0 +1,299 @@
+    };
+    fetchWithCreds({
+      path: `${requestorPath}request`,
+      method: 'POST',
+      body: JSON.stringify(body),
+    }).then(
+      ({ data, status }) => {
+        if (status === 201) {
+          if (data && data.request_id) {
+            // request created, now create a kayako ticket
+            const fullName = `${formValues['First Name']} ${formValues['Last Name']}`;
+            const email = formValues['E-mail Address'];
+            const subject = 'Registration Access Request for Workspace';
+
+            let contents = `Request ID: ${data.request_id}\nEnvironment: ${hostname}`;
+
+            Object.entries(formValues).filter(([key]) => !key.includes('_doNotInclude')).forEach((entry) => {
+              const [key, value] = entry;
+              contents = contents.concat(`\n${key}: ${value}`);
+            });
+            createKayakoTicket(subject, fullName, email, contents, kayakoConfig?.kayakoDepartmentId).then(() => setFormSubmissionStatus({ status: 'success' }),
+              (err) => setFormSubmissionStatus({ status: 'error', text: err.message }));
+          } else {
+            // eslint-disable-next-line no-console
+            console.error('Requestor returns 201 but no request_id in payload body'); // shouldn't get here
+          }
+        } else if (status === 409) {
+          // there is already a request for this user on this study, display a message and disable the button
+          setFormSubmissionButtonDisabled(true);
+          setFormSubmissionStatus({ status: 'warning', text: 'There is already a pending request for workspace access, please wait while we are processing your request.' });
+        } else {
+          // something has gone wrong
+          setFormSubmissionStatus({ status: 'error', text: `Failed to create a request with error code: ${status}. Please try again later. If the error persists, please contact us for help.` });
+        }
+        setReqAccessRequestPending(false);
+      },
+    );
+  };
+
+  const onFinish = (values) => {
+    setReqAccessRequestPending(true);
+    handleRegisterFormSubmission(values);
+  };
+
+  const onReset = () => {
+    form.resetFields();
+  };
+
+  const onRadioChange = (e) => {
+    setRole(e.target.value);
+  };
+
+  if (formSubmissionStatus) {
+    return (
+      <div className='workspace-reg-container'>
+        <div className='workspace-reg-form-container'>
+          {(formSubmissionStatus.status === 'success') ? (
+            <Result
+              status={formSubmissionStatus.status}
+              title='Your access request has been submitted!'
+              subTitle='Thank you for your submission. Requests take up to 1 business day to complete. You will be notified of the status.'
+              extra={[
+                <Link key='discovery' to={'/discovery'}>
+                  <Button>Go To Discovery Page</Button>
```

Comment:
Does it needs a go to discovery page button? Would a go to homepage button be more meaningful?

Reply:
Thanks for pointing thing out. Ideally for BRH, we still need the redirect to be to the `Discovery Page`. But after you pointed that out, I've gone ahead and added this as a [configurable field](https://github.com/uc-cdis/gitops-qa/blob/6cb408012f1faf8e2d30ea6b305362d21359cda0/qa-brh.planx-pla.net/portal/gitops.json#L460-L462) in workspaceConfig

Reply:
This might be a bit over complicated, but for BRH and HEAL, their homepage is indeed the discovery page already, because they are using the ecosystem bundle

Reply:
What do you suggest would be the best way to go? The advantage of making it configurable is, other commons which don't rely on `Discovery Page` can also directly include this feature.